### ISSUE TITLE
Option to force expiration evaluation on startup

### ISSUE THREAD

**Issue by SrBedrock** [GPAddons/GPClaimExpiration#1]
I have thousands of protections that should already be abandoned and I would like to be able to activate an option that would check all protections saved at startup.

**Comment by Jikoo** [comment]
So for clarity, how is GPCE running? Everything seems normal in logs, claims actually being removed? This seems like a one-time-use sort of feature that would be resolved by making sure every claim is considered within the time between restarts. If it's not intended to be one-time-use you're just blocking players from rejoining your server for longer while something that could occur naturally during uptime takes place.

If you urgently need to consider all players, could you try setting percent value to 1000 temporarily or something? That would ensure that all players with claims are considered in 6 minutes, after which you could revert to less-aggressive settings.

If you are seeing claims that should be expired after GPCE finishes a run, please enable debug logging in GP (GPCE piggybacks its abridged log system) and pass me the results, I really want to make sure there's some functional periodic expiration system available for GP.

**Comment by SrBedrock** [comment]
> So for clarity, how is GPCE running? Everything seems normal in logs, claims actually being removed? 

Nothing is being displayed in the log regarding GPClaimExpiration, only when the plugin is activated at the start of the server, should something be displayed when the protections are removed? I will activate griefprevention debug mode to see if something is displayed.

>  This seems like a one-time-use sort of feature that would be resolved by making sure every claim is considered within the time between restarts. If it's not intended to be one-time-use you're just blocking players from rejoining your server for longer while something that could occur naturally during uptime takes place.

This is the idea, something that would be activated once every year during the start of the server for example and not something that would be on all the time.

> If you urgently need to consider all players, could you try setting percent value to 1000 temporarily or something? That would ensure that all players with claims are considered in 6 minutes, after which you could revert to less-aggressive settings.

Like this:
```yml
expiration:
  evaluation:
    rate:
      # Check types: PERCENT, COUNT
      type: PERCENT
      value: 1000
    random: false
  days_per_area:
    250000: 100
    10000: 90
    100: 60
    0: 30
  bypass:
    claim_blocks: -1
    bonus_claim_blocks: -1
    permissions:
      - gpclaimexpiration.persist
  # Command placeholders: $playerName, $playerUUID, $world, $locX, $locY, $locZ
  # Additional delete command placeholders: $claimId, $locMaxX, $locMaxY, $locMaxZ, $area
  claim.commands: []
  pet:
    days: 30
    commands: []
```

> If you are seeing claims that should be expired after GPCE finishes a run, please enable debug logging in GP (GPCE piggybacks its abridged log system) and pass me the results, I really want to make sure there's some functional periodic expiration system available for GP.

As I am not finding anything in the server log that indicates the functioning of GPClaimExpiration, I will activate the function before performing the procedure recommended above.

**Comment by SrBedrock** [comment]
GriefPrevention log: https://gist.github.com/SrBedrock/bf3dc33f885a7c0c8bba6975346feb7f

**Comment by Jikoo** [comment]
Okay, definitely not working correctly then. I probably won't have time to check it out today unfortunately, but that's good info, thank you.

**Comment by Jikoo** [comment]
Found time. Please try [the latest dev build](https://ci.appveyor.com/project/Jikoo/gpclaimexpiration/build/artifacts) and pass me the GP logs.

**Comment by SrBedrock** [comment]
GriefPrevention log: https://gist.github.com/SrBedrock/d90a717625b8a44acf1875e54c55b510

**Comment by Jikoo** [comment]
Okay, I spent like an hour trying to stop scheduled tasks by class and couldn't get it, so I dropped the idea of stopping GP's task for now. Dev build should properly evaluate everyone.

**Comment by SrBedrock** [comment]
GriefPrevention log with version: https://github.com/GPAddons/GPClaimExpiration/commit/7d6e3d42db4b5009c04e61ef028956e627d5688e
https://gist.github.com/SrBedrock/66c07cc911e9e80b0796dba9f198f529

**Comment by Jikoo** [comment]
Looks like it's working, just very slowly:
```
22:38 [GPClaimExpiration] Checking expiration for 567e227e-e661-3185-9e03-0921fc812845
22:38 [GPClaimExpiration] 567e227e-e661-3185-9e03-0921fc812845 is not exempt from expiration.
22:38 [GPClaimExpiration] 567e227e-e661-3185-9e03-0921fc812845 has not been online for 246 days, claims may be eligible to delete.
22:38 [GPClaimExpiration] 3580 has an area of 441 and is eligible for delete
22:38 [GPClaimExpiration] 3608 has an area of 676 and is eligible for delete
22:38 [GPClaimExpiration] Claim 3580 by 567e227e-e661-3185-9e03-0921fc812845 has expired.
22:38 [GPClaimExpiration] Claim 3608 by 567e227e-e661-3185-9e03-0921fc812845 has expired.
```

Could you toss me your config so I can try to figure out why it's not evaluating faster?

**Comment by SrBedrock** [comment]
GriefPrevention Config: https://gist.github.com/SrBedrock/62ec81845dae52e5662a16cb42a42f4b
GPClaimExpiration Config: https://gist.github.com/SrBedrock/0fff28d46981d3b5f26fb3d4d691d039

**Comment by Jikoo** [comment]
Okay, if you would please try out the latest again I've fixed the order of operations and a general gradual slowdown in percent-based mode.

**Comment by SrBedrock** [comment]
It worked! Yesss!
GriefPrevention Log: https://gist.github.com/SrBedrock/8be3c8a6358db26ca0707c99fe3325eb
Jikoo do you have discord?

**Comment by Jikoo** [comment]
Excellent, glad to hear that did the trick.

Re: Discord
Let me preface this by saying that this is a boilerplate reply and I don't mean any offense or personal attack by it.

I do not offer support via any form of instant messaging.
* Instant messaging gets you a lower quality reply.
  * Due to my own limited time, my response speed to IMs is just as slow as on forums.
  * Replying to several lines of content a couple hours after it is posted is much more difficult when using an IM platform.
  * I read things before I have enough free time to formulate a reply. For email/forums/issue trackers this is fine, but unfortunately, for IM it's a nasty habit.  Because there's no way for me to mark messages as unread, I sometimes forget the message exists and never reply at all.
* Instant messaging encourages lower quality interaction.
  * People take the easy route. Why research when they can ask me a question and I'll research it for them?
  * There's social pressure for me to reply faster without properly researching issues.
  * People feel entitled to support when I'm online. This hobby becomes a lot less fun (which in turn only slows down my support) when people demand to know exactly what I'm doing.
* Instant messaging produces fewer useful public resources.
  * IM encourages repeat/duplicate questions because the information is harder (or impossible, depending on settings) to find.

Tl;dr: Yes, I do have Discord, but I don't give it out. If you do really need to contact me privately, I read DMs on Spigot/Bukkit and emails, but for development stuff I prefer to be either tagged in a thread or mentioned in an issue if my input is desired.
